### ISSUE TITLE
failed to create cluster in namespace... failed to start the mons

### ISSUE THREAD

**Issue by pcm32** [rook/rook#1903]
**Is this a bug report or feature request?**
Bug Report

**Deviation from expected behavior:**

After starting operator through helm, and create cluster through kubectl, cluster doesn't seem to be created, operator log repeatedly shows (until giving up):
```
op-mon: start running mons
op-cluster: failed to create cluster in namespace rook. failed to start the mons. failed to initialize ceph cluster info. failed to get cluster info. failed to get mon secrets. secrets "rook-ceph-mon" is forbidden: User "system:serviceaccount:rook:rook-ceph-system" cannot get secrets in the namespace "rook"
```

**Expected behavior:**

Cluster gets created.

**How to reproduce it (minimal and precise):**
```
# Create a rook namespace
# Create operator with helm:
helm install --name rook --namespace rook rook-master/rook-ceph -f rook-config.yaml --version v0.7.0-284.g863c10f
# Create cluster:
kubectl create -f rook-cluster.yaml
```

rook-config.yaml:
```
agent:
  flexVolumeDirPath: "/var/lib/kubelet/volume-plugins"
```

rook-cluster.yaml:
```
---
apiVersion: ceph.rook.io/v1beta1
kind: Cluster
metadata:
  name: rook-ceph
  namespace: rook
spec:
  dataDirHostPath: /var/lib/rook
  dashboard:
    enabled: true
  serviceAccount: rook-ceph-cluster
  resources:
    limits:
      cpu: "4"
      memory: "8Gi"
  storage:
    useAllNodes: false
    useAllDevices: false
    nodes:
    - name: "solr-scale-k8s-node-nf-1"
    - name: "solr-scale-k8s-node-nf-2"
    - name: "solr-scale-k8s-node-nf-3"
    storeConfig:
      storeType: filestore
      databaseSizeMB: 1024
      journalSizeMB: 1024
```

**Environment**:
* OS (e.g. from /etc/os-release):
Ubuntu 16.04.4 LTS
* Kernel (e.g. `uname -a`):
Linux k8s-master-1 4.4.0-122-generic #146-Ubuntu SMP Mon Apr 23 15:34:04 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
* Cloud provider or hardware configuration:
OpenStack
* Rook version (use `rook version` inside of a Rook Pod):
v0.7.0-284.g863c10f (from helm)
* Kubernetes version (use `kubectl version`):
1.9.5
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
Kubespray v2.4.0
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/Rook/master/toolbox.html)):
Not sure I get this far.

* Additionals:
PodSecurityPolicy Admission Controller is not activated as far as I can see.

Thanks!

**Comment by galexrt** [comment]
@pcm32 You have not created the required RBAC roles for a cluster, please try again with this manifest: https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/cluster.yaml
The manifest contains RBAC roles, ServiceAccounts and so on which are needed for the operator to access the namespace in which the Rook Cluster was created.

**Comment by pcm32** [comment]
Thanks for the prompt answer! So these are not created then by the Helm chart? Documentation seems to lead users to go straight from helm chart installation to cluster setup (or then I missed something).

I'll try what you suggest and report back.

**Comment by pcm32** [comment]
Ok, chugging along apparently! but I guess documentation needs to point out that these RBAC related objects need to be created by the user [here](https://rook.io/docs/rook/v0.7/quickstart.html) in between the part `You can also deploy the operator with the Rook Helm Chart.` and section `Create a Rook Cluster`.

Albeit different, after a number of other messages, operator logs read now:
```
2018-07-18 10:01:22.427889 I | op-osd: 0/3 node(s) completed osd provisioning
2018-07-18 10:02:22.429096 I | op-osd: waiting on orchestration status update from 3 remaining nodes
2018-07-18 10:03:22.429338 I | op-osd: waiting on orchestration status update from 3 remaining nodes
2018-07-18 10:04:22.429602 I | op-osd: waiting on orchestration status update from 3 remaining nodes
2018-07-18 10:05:22.429806 I | op-osd: waiting on orchestration status update from 3 remaining nodes
2018-07-18 10:06:22.430113 I | op-osd: waiting on orchestration status update from 3 remaining nodes
2018-07-18 10:07:22.430352 I | op-osd: waiting on orchestration status update from 3 remaining nodes
2018-07-18 10:08:22.430603 I | op-osd: waiting on orchestration status update from 3 remaining nodes
2018-07-18 10:09:22.430886 I | op-osd: waiting on orchestration status update from 3 remaining nodes
```

is this fine? I can open another issue of course.

**Comment by pcm32** [comment]
Will close this as the solution by @galexrt sorted my original issue.
